portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -44,16 +44,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Query query = new Query(ENTITY_TITLE);
     PreparedQuery results = datastore.prepare(query);
     
+    int userChoice = getUserChoice(request);
+    if(userChoice == -1){
+        response.setContentType("text/html");
+        response.getWriter().println("Please enter an integer between 1 and 3");
+        return;
+    }
+      
     final List<Comment> comments = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
-      String comment = (String) entity.getProperty(ENTITY_PROPERTY_KEY);
-      long id = entity.getKey().getId();
-      Comment text = new Comment(id,comment);
-      comments.add(text);
+        String comment = (String) entity.getProperty(ENTITY_PROPERTY_KEY);
```

Comment:
instead of creating a list from all results and then creating a sublist of that later, just stop adding to `comments` if the size is larger than the max.

`if(comments.size() >= userChoice) { break; }`

The break will exit the for loop

Reply:
Oh yes, you're right, thank you!